File: content/04/code/java/10--step10-delete-unused-stepdef/shouty/src/main/java/shouty/Network.java
package shouty;

import java.util.ArrayList;
import java.util.List;

public class Network {
    private final List<Person> listeners = new ArrayList<Person>();

    public void subscribe(Person person) {
        listeners.add(person);
    }

    public void broadcast(String message) {
        for (Person listener : listeners) {
            listener.hear(message);
        }
    }
}

File: content/bdd-with-cucumber/04/code/dotnet/14--step14-replace-hardcoded-sean-instance/Shouty/Shouty/Network.cs
using System;
using System.Collections.Generic;

namespace Shouty
{
    public class Network
    {
        private readonly List<Person> listeners = new List<Person>();

        public virtual void Subscribe(Person person)
        {
            listeners.Add(person);
        }

        public virtual void Broadcast(string message)
        {
            foreach (var listener in listeners)
            {
                listener.Hear(message);
            }
        }
    }
}
